### PR TITLE
makes sourcemap generation optional

### PR DESCRIPTION
**Pull request by orthagonal**
'on' will generate a sourcemap file and embed a ref to it at the end of
the output file
'off' will not generate a sourcemap
'inline' will generate a sourcemap and embed it directly at the end of
the output file

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -38,6 +38,10 @@ const argv = yargs
   describe: 'a path to your configuration files',
   default: path.join(process.cwd(), 'clientkit')
 })
+.option('sourcemap', {
+  describe: 'determines how/whether css/js sourcemaps are generated, can be one of "on", "off" or "inline"',
+  default: 'on'
```
Comment (jgallen23) [inline app.js]: can we default to the option in the conf?

File: lib/removeSourceMap.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// removes embedded source maps in output css/js files:
+module.exports = (text, type) => {
```
Comment (jgallen23) [inline lib/removeSourceMap.js]: why do we need this?  if we don't run through sourcemap process or enable debugging, this shouldn't get added, correct?

File: tasks/script.js [hunk 3]
```diff
@@ -61,12 +71,16 @@ module.exports = function(conf, base, outputName, input) {
   if (conf.core.minify) {
     currentTransform = currentTransform.transform(uglifyify, { global: true });
   }
-  currentTransform
+  const stream = currentTransform
```
Comment (jgallen23) [inline tasks/script.js]: debug needs to be set to false above if sourcemaps are off.

File: tasks/script.js [hunk 4]
```diff
@@ -61,12 +71,16 @@ module.exports = function(conf, base, outputName, input) {
...
   .bundle()
   .on('error', function (err) {
     log(['error'], err);
     this.emit('end');
-  })
-  .pipe(exorcist(`${output}.map`))
-  .pipe(fileStream);
+  });
+  if (conf.sourcemap === 'on') {
+    stream.pipe(exorcist(`${output}.map`))
```
Comment (jgallen23) [inline tasks/script.js]: does inline work for javascript?

File: app.js (hunk 1 above)
Comment (jgallen23) [inline app.js]: actually, lets just set it in the config yaml instead of cli arg

File: lib/removeSourceMap.js (hunk 2 above)
Comment (orthagonal) [inline lib/removeSourceMap.js]: I tried to make this happen, but neither postcss and browserify seem to support a basic option to stop them from planting a sourcemap tag at the end of the css/js file.  Just the power to either embed the sourcemap directly or as a link to the external file.

File: tasks/script.js (hunk 3 above)
Comment (orthagonal) [inline tasks/script.js]: Got it.

File: tasks/script.js (hunk 4 above)
Comment (orthagonal) [inline tasks/script.js]: Yup.

**Pull request closed without merging**